ts/DataGrid/DataGrid2/DataGridRow.ts

```diff
@@ -0,0 +1,201 @@
+/* *
+ *
+ *  Data Grid class
+ *
+ *  (c) 2020-2024 Highsoft AS
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ *  Authors:
+ *  - Dawid Dragula
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import DataGridCell from './DataGridCell.js';
+import DataGridTable from './DataGridTable.js';
+import Globals from './Globals.js';
+import Utils from './Utils.js';
+import DataGrid from './DataGrid.js';
+
+const { makeHTMLElement } = Utils;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * Represents a row in the data grid.
+ */
+class DataGridRow {
+
+    /* *
+    *
+    *  Properties
+    *
+    * */
+
+    /**
+     * The cells of the row.
+     */
+    public cells: DataGridCell[] = [];
+
+    /**
+     * The HTML element of the row.
+     */
+    public htmlElement: HTMLTableRowElement;
+
+    /**
+     * The dataGrid instance which the row belongs to.
+     */
+    public dataGrid: DataGrid;
+
+    /**
+     * The index of the row in the data table.
+     */
+    public index: number;
+
+    /**
+     * Whether the row is destroyed.
+     */
+    public destroyed: boolean = false;
```

Comment:
`destroyed` for me seems permanent and completed, I would use something like removed/hidden/notVisibleInViewport?